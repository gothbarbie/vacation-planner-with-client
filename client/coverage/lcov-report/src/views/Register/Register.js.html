<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/views/Register/Register.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/views/Register</a> Register.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">13.04% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/23</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">10% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.86% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">13.04% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/23</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
import React, { Component } from 'react'
import { connect } from 'react-redux'
&nbsp;
import H1 from '../../components/H1'
import Form from '../../components/Form'
import Input from '../../components/FormInput'
import ButtonLink from '../../components/ButtonLink'
import Icon from '../../components/Icon'
import * as actions from './registerActions'
&nbsp;
import './Register.css'
&nbsp;
type Props = {
  registerUser: Function,
}
&nbsp;
type State = {
  firstName: {
    value: string,
    touched: boolean,
  },
  lastName: {
    value: string,
    touched: boolean,
  },
  email: {
    value: string,
    touched: boolean,
  },
  password: {
    value: string,
    touched: boolean,
  },
  passwordConfirm: {
    value: string,
    touched: boolean,
  },
  errors: {
    firstName: string,
    lastName: string,
    email: string,
    password: string,
    passwordConfirm: string,
  },
}
&nbsp;
export class Register extends Component&lt;Props, State&gt; {
  state = {
    firstName: {
      value: '',
      touched: false,
    },
    lastName: {
      value: '',
      touched: false,
    },
    email: {
      value: '',
      touched: false,
    },
    password: {
      value: '',
      touched: false,
    },
    passwordConfirm: {
      value: '',
      touched: false,
    },
    errors: {
      firstName: '',
      lastName: '',
      email: '',
      password: '',
      passwordConfirm: '',
    },
  }
&nbsp;
  handleSubmit = <span class="fstat-no" title="function not covered" >(e</span>vent: Event) =&gt; {
<span class="cstat-no" title="statement not covered" >    event.preventDefault()</span>
<span class="cstat-no" title="statement not covered" >    this.props.registerUser(this.state)</span>
  }
&nbsp;
  handleChange = <span class="fstat-no" title="function not covered" >(e</span>vent?: SyntheticInputEvent&lt;any&gt;) =&gt; {
<span class="cstat-no" title="statement not covered" >    if (event &amp;&amp; event.target) {</span>
      const target = <span class="cstat-no" title="statement not covered" >event.target</span>
      const value = <span class="cstat-no" title="statement not covered" >target.type === 'checkbox' ? target.checked : target.value</span>
      const name = <span class="cstat-no" title="statement not covered" >target.name</span>
<span class="cstat-no" title="statement not covered" >      this.setState({ [name]: { ...this.state[name], value } })</span>
    }
  }
&nbsp;
  handleBlur = (field: string) =&gt; <span class="fstat-no" title="function not covered" >(e</span>vent: SyntheticFocusEvent&lt;any&gt;) =&gt; {
<span class="cstat-no" title="statement not covered" >    this.setState({</span>
      [field]: { ...this.state[field], touched: true },
    })
<span class="cstat-no" title="statement not covered" >    this.validateForm()</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  va</span>lidateForm () {
    const errors = <span class="cstat-no" title="statement not covered" >{}</span>
&nbsp;
    const email = <span class="cstat-no" title="statement not covered" >this.state.email.value</span>
    const password = <span class="cstat-no" title="statement not covered" >this.state.password.value</span>
    const passwordConfirm = <span class="cstat-no" title="statement not covered" >this.state.passwordConfirm.value</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    errors.email = !email.length &amp;&amp; 'Email is required'</span>
<span class="cstat-no" title="statement not covered" >    errors.password = !password.length &amp;&amp; 'Password is required'</span>
<span class="cstat-no" title="statement not covered" >    errors.passwordConfirm =</span>
      !passwordConfirm.length &amp;&amp; 'Confirm Password is required'
<span class="cstat-no" title="statement not covered" >    if (!errors.passwordConfirm) {</span>
<span class="cstat-no" title="statement not covered" >      errors.passwordConfirm =</span>
        password !== passwordConfirm &amp;&amp; 'Passwords must match'
    }
&nbsp;
<span class="cstat-no" title="statement not covered" >    this.setState({</span>
      errors: { ...this.state.errors, errors },
    })
  }
&nbsp;
  showErrors = (field: string) =&gt; {
    <span class="missing-if-branch" title="if path not taken" >I</span>if (this.state.errors[field] &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >this.state[field].touched)</span> {
<span class="cstat-no" title="statement not covered" >      return this.state.errors[field]</span>
    }
  }
&nbsp;
  render () {
    return (
      &lt;section className="register"&gt;
        &lt;H1&gt;New User&lt;/H1&gt;
        
        &lt;div className="register__inner"&gt;
          &lt;div className="register__email"&gt;
            &lt;Form onSubmit={this.handleSubmit} title="Enter your credentials"&gt;
              &lt;div className="register__columns"&gt;
                &lt;Input
                  error={this.showErrors('firstName')}
                  handleBlur={this.handleBlur('firstName')}
                  handleChange={this.handleChange}
                  label="First Name"
                  name="firstName"
                  value={this.state.firstName.value}
                /&gt;
                &lt;Input
                  error={this.showErrors('lastName')}
                  handleBlur={this.handleBlur('lastName')}
                  handleChange={this.handleChange}
                  label="Family Name"
                  name="lastName"
                  value={this.state.lastName.value}
                /&gt;
                &lt;Input
                  error={this.showErrors('email')}
                  handleBlur={this.handleBlur('email')}
                  handleChange={this.handleChange}
                  label="E-mail"
                  name="email"
                  required
                  type="email"
                  value={this.state.email.value}
                /&gt;
                &lt;Input
                  error={this.showErrors('password')}
                  handleBlur={this.handleBlur('password')}
                  handleChange={this.handleChange}
                  label="Password"
                  name="password"
                  required
                  type="password"
                  value={this.state.password.value}
                /&gt;
                &lt;Input
                  error={this.showErrors('passwordConfirm')}
                  handleBlur={this.handleBlur('passwordConfirm')}
                  handleChange={this.handleChange}
                  label="Confirm Password"
                  name="passwordConfirm"
                  required
                  type="password"
                  value={this.state.passwordConfirm.value}
                /&gt;
              &lt;/div&gt;
            &lt;/Form&gt;
          &lt;/div&gt;
          &lt;div className="register__or"&gt;Or...&lt;/div&gt;
          &lt;div className="register__google"&gt;
            &lt;ButtonLink secondary url="/auth/google"&gt;
              &lt;Icon isBrand marginRight name="google-plus" /&gt;Register with
              Google
            &lt;/ButtonLink&gt;
          &lt;/div&gt;
        &lt;/div&gt;
      &lt;/section&gt;
    )
  }
}
&nbsp;
export default connect(null, actions)(Register)
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Mar 11 2018 12:53:11 GMT+0100 (CET)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
