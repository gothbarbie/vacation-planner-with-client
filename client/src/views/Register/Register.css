.register__columns {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: auto;
  grid-gap: 0 1rem;
  justify-content: space-around;
}

.register__or {
  font-family: 'Avenir Next', sans-serif;
}

.register__google {
  display: flex;
  justify-content: center;
  align-items: center;
}
