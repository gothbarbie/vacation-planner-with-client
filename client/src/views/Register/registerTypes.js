export const REGISTER_USER = 'register/register_user'
