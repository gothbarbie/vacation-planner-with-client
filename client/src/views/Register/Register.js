// @flow
import React, { Component } from 'react'
import { graphql } from 'react-apollo'
import gql from 'graphql-tag'

import PageWrapper from '../../components/PageWrapper'
import ThreeColumnsWrapper from '../../components/ThreeColumnsWrapper'
import H1 from '../../components/H1'
import Form from '../../components/Form'
import Input from '../../components/FormInput'
import ButtonLink from '../../components/ButtonLink'
import Icon from '../../components/Icon'

import './Register.css'

type Props = {
  mutate: Function,
}

type State = {
  firstName: {
    value: string,
    touched: boolean,
  },
  lastName: {
    value: string,
    touched: boolean,
  },
  email: {
    value: string,
    touched: boolean,
  },
  password: {
    value: string,
    touched: boolean,
  },
  passwordConfirm: {
    value: string,
    touched: boolean,
  },
  errors: {
    firstName: string,
    lastName: string,
    email: string,
    password: string,
    passwordConfirm: string,
  },
}

export class Register extends Component<Props, State> {
  state = {
    firstName: {
      value: '',
      touched: false,
    },
    lastName: {
      value: '',
      touched: false,
    },
    email: {
      value: '',
      touched: false,
    },
    password: {
      value: '',
      touched: false,
    },
    passwordConfirm: {
      value: '',
      touched: false,
    },
    errors: {
      firstName: '',
      lastName: '',
      email: '',
      password: '',
      passwordConfirm: '',
    },
  }

  handleSubmit = (event: Event) => {
    event.preventDefault()
    const { firstName, lastName, email, password } = this.state
    this.props.mutate({
      variables: {
        firstName: firstName.value,
        lastName: lastName.value,
        email: email.value,
        password: password.value,
      },
    })
  }

  handleChange = (event?: SyntheticInputEvent<any>) => {
    if (event && event.target) {
      const target = event.target
      const value = target.type === 'checkbox' ? target.checked : target.value
      const name = target.name
      this.setState({ [name]: { ...this.state[name], value } })
    }
  }

  handleBlur = (field: string) => (event: SyntheticFocusEvent<any>) => {
    this.setState({
      [field]: { ...this.state[field], touched: true },
    })
    this.validateForm()
  }

  validateForm () {
    const errors = {}

    const email = this.state.email.value
    const password = this.state.password.value
    const passwordConfirm = this.state.passwordConfirm.value

    errors.email = !email.length && 'Email is required'
    errors.password = !password.length && 'Password is required'
    errors.passwordConfirm =
      !passwordConfirm.length && 'Confirm Password is required'
    if (!errors.passwordConfirm) {
      errors.passwordConfirm =
        password !== passwordConfirm && 'Passwords must match'
    }

    this.setState({
      errors: { ...this.state.errors, errors },
    })
  }

  showErrors = (field: string) => {
    if (this.state.errors[field] && this.state[field].touched) {
      return this.state.errors[field]
    }
  }

  render () {
    return (
      <PageWrapper>
        <H1>New User</H1>

        <ThreeColumnsWrapper>
          <div className="register__email">
            <Form onSubmit={this.handleSubmit} title="Enter your credentials">
              <div className="register__columns">
                <Input
                  error={this.showErrors('firstName')}
                  handleBlur={this.handleBlur('firstName')}
                  handleChange={this.handleChange}
                  label="First Name"
                  name="firstName"
                  value={this.state.firstName.value}
                />
                <Input
                  error={this.showErrors('lastName')}
                  handleBlur={this.handleBlur('lastName')}
                  handleChange={this.handleChange}
                  label="Family Name"
                  name="lastName"
                  value={this.state.lastName.value}
                />
                <Input
                  error={this.showErrors('email')}
                  handleBlur={this.handleBlur('email')}
                  handleChange={this.handleChange}
                  label="E-mail"
                  name="email"
                  required
                  type="email"
                  value={this.state.email.value}
                />
                <Input
                  error={this.showErrors('password')}
                  handleBlur={this.handleBlur('password')}
                  handleChange={this.handleChange}
                  label="Password"
                  name="password"
                  required
                  type="password"
                  value={this.state.password.value}
                />
                <Input
                  error={this.showErrors('passwordConfirm')}
                  handleBlur={this.handleBlur('passwordConfirm')}
                  handleChange={this.handleChange}
                  label="Confirm Password"
                  name="passwordConfirm"
                  required
                  type="password"
                  value={this.state.passwordConfirm.value}
                />
              </div>
            </Form>
          </div>
          <div className="register__or">Or...</div>
          <div className="register__google">
            <ButtonLink secondary url="/auth/register/google">
              <Icon isBrand marginRight name="google-plus" />Register with
              Google
            </ButtonLink>
          </div>
        </ThreeColumnsWrapper>
      </PageWrapper>
    )
  }
}

const mutation = gql`
  mutation AddUser(
    $email: String!
    $firstName: String
    $lastName: String
    $password: String!
  ) {
    addUser(
      email: $email
      firstName: $firstName
      lastName: $lastName
      password: $password
    ) {
      id
    }
  }
`

export default graphql(mutation)(Register)
