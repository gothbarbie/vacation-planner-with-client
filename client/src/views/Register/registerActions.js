import { REGISTER_USER } from './registerTypes'

export const registerUser = (user) => async dispatch => {

  console.log('registerUser', user)

  dispatch({
    type: REGISTER_USER, 
    payload: {},
  })
}
