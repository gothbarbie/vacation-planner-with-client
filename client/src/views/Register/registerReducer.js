import { REGISTER_USER } from './registerTypes'

export default function (state = {}, action) {
  switch (action.type) {
    case REGISTER_USER:
      return action.payload
    default:
      return state
  }
}
