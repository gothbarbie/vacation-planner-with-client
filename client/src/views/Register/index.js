// @flow
import React, { Component } from 'react'
import { connect } from 'react-redux'

import H1 from '../../components/common/H1'
import Form from '../../components/common/Form'
import Input from '../../components/common/Form/Input'
import { ButtonLink } from '../../components/common/Button'
import Icon from '../../components/common/Icon'
import * as actions from './registerActions'

import './Register.css'

type Props = {
  registerUser: Function,
}

type State = {
  firstName: {
    value: string,
    touched: boolean,
  },
  lastName: {
    value: string,
    touched: boolean,
  },
  email: {
    value: string,
    touched: boolean,
  },
  password: {
    value: string,
    touched: boolean,
  },
  passwordConfirm: {
    value: string,
    touched: boolean,
  },
  errors: {
    firstName: string,
    lastName: string,
    email: string,
    password: string,
    passwordConfirm: string,
  },
}

class Register extends Component<Props, State> {
  state = {
    firstName: {
      value: '',
      touched: false,
    },
    lastName: {
      value: '',
      touched: false,
    },
    email: {
      value: '',
      touched: false,
    },
    password: {
      value: '',
      touched: false,
    },
    passwordConfirm: {
      value: '',
      touched: false,
    },
    errors: {
      firstName: '',
      lastName: '',
      email: '',
      password: '',
      passwordConfirm: '',
    },
  }

  handleSubmit = event => {
    event.preventDefault()
    this.props.registerUser(this.state)
  }

  handleChange = (event : Event) => {
    if (event && event.target) {
      const target = event.target
      const value = target.type === 'checkbox' ? target.checked : target.value
      const name = target.name
      this.setState({ [name]: { ...this.state[name], value } })
    }
  }

  handleBlur = (field: string) => (event: Event) => {
    this.setState({
      [field]: { ...this.state[field], touched: true },
    })
    this.validateForm()
  }

  validateForm () {
    const errors = {}

    const email = this.state.email.value
    const password = this.state.password.value
    const passwordConfirm = this.state.passwordConfirm.value

    errors.email = !email.length && 'Email is required'
    errors.password = !password.length && 'Password is required'
    errors.passwordConfirm =
      !passwordConfirm.length && 'Confirm Password is required'
    if (!errors.passwordConfirm) {
      errors.passwordConfirm =
        password !== passwordConfirm && 'Passwords must match'
    }

    this.setState({
      errors: { ...this.state.errors, errors },
    })
  }

  showErrors = (field: string) => {
    if (this.state.errors[field] && this.state[field].touched) {
      return this.state.errors[field]
    }
  }

  render () {
    return (
      <section className="register">
        <H1>New User</H1>
        <div className="register__inner">
          <div className="register__email">
            <Form onSubmit={this.handleSubmit} title="Enter your credentials">
              <div className="register__columns">
                <Input
                  error={this.showErrors('firstName')}
                  handleBlur={this.handleBlur('firstName')}
                  handleChange={this.handleChange}
                  label="First Name"
                  name="firstName"
                  value={this.state.firstName.value}
                />
                <Input
                  error={this.showErrors('lastName')}
                  handleBlur={this.handleBlur('lastName')}
                  handleChange={this.handleChange}
                  label="Family Name"
                  name="lastName"
                  value={this.state.lastName.value}
                />
                <Input
                  error={this.showErrors('email')}
                  handleBlur={this.handleBlur('email')}
                  handleChange={this.handleChange}
                  label="E-mail"
                  name="email"
                  required
                  type="email"
                  value={this.state.email.value}
                />
                <Input
                  error={this.showErrors('password')}
                  handleBlur={this.handleBlur('password')}
                  handleChange={this.handleChange}
                  label="Password"
                  name="password"
                  required
                  type="password"
                  value={this.state.password.value}
                />
                <Input
                  error={this.showErrors('passwordConfirm')}
                  handleBlur={this.handleBlur('passwordConfirm')}
                  handleChange={this.handleChange}
                  label="Confirm Password"
                  name="passwordConfirm"
                  required
                  type="password"
                  value={this.state.passwordConfirm.value}
                />
              </div>
            </Form>
          </div>
          <div className="register__or">Or...</div>
          <div className="register__google">
            <ButtonLink secondary url="/auth/google">
              <Icon isBrand marginRight name="google-plus" />Register with
              Google
            </ButtonLink>
          </div>
        </div>
      </section>
    )
  }
}

export default connect(null, actions)(Register)
