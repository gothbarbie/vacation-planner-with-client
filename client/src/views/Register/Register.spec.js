import React from 'react'
import { shallow } from 'enzyme'

import {Â Register } from './Register'

describe('views/Register', () => {
  const register = shallow(<Register />)

  it('renders', () => {
    expect(register).toMatchSnapshot()
  })
})
