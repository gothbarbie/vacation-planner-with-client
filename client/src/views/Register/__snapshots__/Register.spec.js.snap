// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/Register renders 1`] = `
<section
  className="register"
>
  <H1>
    New User
  </H1>
  <div
    className="register__inner"
  >
    <div
      className="register__email"
    >
      <Component
        onSubmit={[Function]}
        title="Enter your credentials"
      >
        <div
          className="register__columns"
        >
          <Input
            handleBlur={[Function]}
            handleChange={[Function]}
            label="First Name"
            name="firstName"
            value=""
          />
          <Input
            handleBlur={[Function]}
            handleChange={[Function]}
            label="Family Name"
            name="lastName"
            value=""
          />
          <Input
            handleBlur={[Function]}
            handleChange={[Function]}
            label="E-mail"
            name="email"
            required={true}
            type="email"
            value=""
          />
          <Input
            handleBlur={[Function]}
            handleChange={[Function]}
            label="Password"
            name="password"
            required={true}
            type="password"
            value=""
          />
          <Input
            handleBlur={[Function]}
            handleChange={[Function]}
            label="Confirm Password"
            name="passwordConfirm"
            required={true}
            type="password"
            value=""
          />
        </div>
      </Component>
    </div>
    <div
      className="register__or"
    >
      Or...
    </div>
    <div
      className="register__google"
    >
      <Component
        secondary={true}
        url="/auth/google"
      >
        <Icon
          isBrand={true}
          marginRight={true}
          name="google-plus"
        />
        Register with Google
      </Component>
    </div>
  </div>
</section>
`;
